gap/digraph.gi

```diff
@@ -381,6 +381,73 @@ function(graph)
   return StructuralCopy(graph!.vertexlabels);
 end);
 
+# edge labels
+InstallMethod(DigraphEdgeLabel, "for a digraph, a pos int, and a pos int",
+[IsDigraph, IsPosInt, IsPosInt],
+function(graph, i, j)
+    local p;
+    DIGRAPHS_InitEdgeLabels(graph);
+    p := Position(graph!.adj[i], j);
+    if p <> fail then
+        return ShallowCopy(graph!.edgelabels[i][p]);
+    else
+        ErrorNoReturn("DigraphEdgeLabel: vertex"
+                     , "[ ", i, ", ", j, " ] is not and edge,");
+    fi;
+end);
+
+InstallMethod(DigraphEdgeLabels, "for a digraph",
+[IsDigraph],
+function(graph)
+    DIGRAPHS_InitEdgeLabels(graph);
+    return StructuralCopy(graph!.edgelabels);
+end);
+
+InstallMethod(SetDigraphEdgeLabel,
+              "for a digraph, a pos int, a pos int, and an object",
+[IsDigraph, IsPosInt, IsPosInt, IsObject],
+function(graph, i, j, label)
+    local p;
+
+    DIGRAPHS_InitEdgeLabels(graph);
+    p := Position(graph!.adj[i], j);
+    if p <> fail then
+        graph!.edgelabels[i][p] := label;
+    else
+        ErrorNoReturn("SetDigraphEdgeLabel: [ ",i,", ", j, " ] is"
+                     , " not an edge");
+    fi;
+end);
+
+InstallMethod(SetDigraphEdgeLabels, "for a digraph, and a list",
+[IsDigraph, IsList],
+function(graph, labels)
+    local i;
+    if Length(labels) = DigraphNrVertices(graph) and
+       ForAll([1..Length(labels)]
+             , i -> Length(labels[i]) = OutDegreeOfVertex(graph, i)) then
+        graph!.edgelabels := List(labels, ShallowCopy);
+    else
+        ErrorNoReturn("SetDigraphEdgeLabels: labels list has wrong shape,"
+                     ," it is required to have the same shape as the"
+                     ," return value of OutNeighbours(<graph>)");
+    fi;
+end);
+
+InstallMethod(SetDigraphEdgeLabels, "for a digraph, and a function",
+[IsDigraph, IsFunction],
+function(graph, wtf)
+    local i,j;
+
+    DIGRAPHS_InitEdgeLabels(graph);
+
```

Comment:
Over the new three lines, I'd prefer it if you didn't access `graph!.adj` directly, and used `adj := OutNeighbours(graph);` to get access to this information instead.